### PR TITLE
AEROGEAR-7670 & AEROGEAR-7673

### PR DESCRIPTION
**Pull request by secondsun**
This will provision the oauth proxy and the mobile-client-service image from docker hub.

After apb build and apb push you should be able to deploy the service from the catalog, navigate to the route, and log in with the oauth proxy.

### CODE REVIEW

**Comment by aidenkeating** [comment]
:eyes:

**Review by wei-lee — CHANGES_REQUESTED** [review]
Getting this error when I try it locally:

```



TASK [provision-mcs-server-apb : Give default user admin permissions] **********
  | fatal: [localhost]: FAILED! => {"changed": true, "cmd": "oc policy add-role-to-user admin system:serviceaccount:myproject:oauth-proxy -n myproject", "delta": "0:00:00.678129", "end": "2018-07-25 11:14:59.240348", "msg": "non-zero return code", "rc": 1, "start": "2018-07-25 11:14:58.562219", "stderr": "Error from server (Forbidden): rolebindings.authorization.openshift.io is forbidden: User \"system:serviceaccount:test-mcs-server-apb-prov-mfjpt:bundle-e9913e24-d451-46d7-8147-ac3746a3865e\" cannot list rolebindings.authorization.openshift.io in the namespace \"myproject\": User \"system:serviceaccount:test-mcs-server-apb-prov-mfjpt:bundle-e9913e24-d451-46d7-8147-ac3746a3865e\" cannot list rolebindings.authorization.openshift.io in project \"myproject\"", "stderr_lines": ["Error from server (Forbidden): rolebindings.authorization.openshift.io is forbidden: User \"system:serviceaccount:test-mcs-server-apb-prov-mfjpt:bundle-e9913e24-d451-46d7-8147-ac3746a3865e\" cannot list rolebindings.authorization.openshift.io in the namespace \"myproject\": User \"system:serviceaccount:test-mcs-server-apb-prov-mfjpt:bundle-e9913e24-d451-46d7-8147-ac3746a3865e\" cannot list rolebindings.authorization.openshift.io in project \"myproject\""], "stdout": "", "stdout_lines": []}
```

Also should the Dockerfile be regenerated? After I ran `apb build` the Dockerfile is changed again.

**Comment by wei-lee** [comment]
Never mind, it is a problem with my local ASB setup.

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+cGVyc2lzdGVudCBzdG9yYWdlIGFuZCBhdXRoZW50aWNhdGlvbiB0aHJvdWdoIE9wZW5zaGlmdCIK\
+ICBpbWFnZVVybDogImh0dHBzOi8vcGJzLnR3aW1nLmNvbS9wcm9maWxlX2ltYWdlcy8xNzk0NDQw\
+MDA1L2Flcm9nZWFyX2ljb24tMV80MDB4NDAwLnBuZyIKICBkb2N1bWVudGF0aW9uVXJsOiAiaHR0\
+cHM6Ly9kb2NzLmFlcm9nZWFyLm9yZy9leHRlcm5hbC9hcGIvbWNzLXNlcnZlci5odG1sIgogIHBy\
+b3ZpZGVyRGlzcGxheU5hbWU6ICJSZWQgSGF0LCBJbmMuIgogIGRlcGVuZGVuY2llczogCiAgICAg\
+IC0gImRvY2tlci5pby9vcGVuc2hpZnQvb2F1dGgtcHJveHk6djEuMS4wIgogICAgICAtICJkb2Nr\
+ZXIuaW8vYWVyb2dlYXIvbW9iaWxlLWNsaWVudC1zZXJ2aWNlOmxhdGVzdCIKICBzZXJ2aWNlTmFt\
+ZTogbWNzLXNlcnZlcgogIGJpbmRpbmdzTGltaXQ6IDIKcGxhbnM6CiAgLSBuYW1lOiBkZWZhdWx0\
+CiAgICBkZXNjcmlwdGlvbjogUGVyc2lzdGVudCBkZXBsb3ltZW50IG9mIG1jcy1zZXJ2ZXIKICAg\
+IGZyZWU6IFRydWUKICAgIHBhcmFtZXRlcnM6IFtdCg=="
```
Comment (wei-lee) [inline Dockerfile]: when I ran `apb build`, the Dockerfile is changed again. Should this file be regenerated?

File: roles/provision-mcs-server-apb/tasks/provision-service-server.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      - containerPort: '{{ mobile_client_service_proxy_port }}'
+        name: public
+      args:
+      - --provider=openshift
+      - --openshift-service-account={{ proxy_serviceaccount_name }}
+      - --upstream=http://localhost:{{ mobile_client_service_port }}
+      - >-
+        --openshift-sar={"namespace":"{{ namespace }}","resource":"deploymentconfigs","name":"mcs-server","verb":"update"}
+      - --http-address=0.0.0.0:{{ mobile_client_service_proxy_port }}
+      - --skip-auth-regex=/rest/sender,/rest/registry/device,/rest/prometheus/metrics,/rest/auth/config
```
Comment (wei-lee) [inline roles/provision-mcs-server-apb/tasks/provision-service-server.yml]: I don't think this are required for this service.

**Review by wei-lee — CHANGES_REQUESTED** [review]
I can provision the service from service catalog, and perform login using my openshift credentials. Although after login I am getting an error but after checking the logs it looks like it has something to do my local cluster setup.

I think we just need to regenerate the dockerfile and update the config for the proxy

File: Dockerfile (hunk 1 above)
Comment (secondsun) [reply Dockerfile]: Yes

**Comment by secondsun** [comment]
@wei-lee apb build regenerates the dockerfile each time.  Not really sure what do do about it.

**Comment by wei-lee** [comment]
@secondsun Yes, it will regenerate the dockerfile, but I thought it shouldn't change anything unless the apb.yaml file has changed?

**Comment by secondsun** [comment]
@wei-lee Ahhh right.  I've been changing the display name so I know when "push" gets finished.  I may have changed a whitespace.

**Comment by secondsun** [comment]
@wei-lee Should be fixed now.

File: roles/provision-mcs-server-apb/tasks/provision-service-server.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      image: '{{ proxy_image }}:{{ proxy_image_tag }}'
+      imagePullPolicy: Always
+      ports:
+      - containerPort: '{{ mobile_client_service_proxy_port }}'
+        name: public
+      args:
+      - --provider=openshift
+      - --openshift-service-account={{ proxy_serviceaccount_name }}
+      - --upstream=http://localhost:{{ mobile_client_service_port }}
+      - --openshift-sar={"namespace":"{{ namespace }}","resource":"deploymentconfigs","name":"mcs-server","verb":"update"}
```
Comment (wei-lee) [inline roles/provision-mcs-server-apb/tasks/provision-service-server.yml:39]: we can leave the option like this for now, but we probably will update it later.

**Review by wei-lee — APPROVED** [review]

**Pull request merged**